### ISSUE TITLE
Hash router issue with deep paths (0.3.1)

### ISSUE THREAD

**Issue by fireph** [solidjs/solid-app-router#89]
I posted ~5 days ago with an issue around hash routing not working (which has since been fixed in 0.3.1) but I am still running into a weird edge case issue where 0.3.1 behavior differs from 0.2.1. Below is my routing setup:

```
<Router source={hashIntegration()}>
  <App />
</Router>
```

routes:
```
<Routes>
  <Route path="/" element={<div>Home</div>} />
  <Route path="/practice" element={<div>Practice</div>} />
  <Route path="/*all" element={<NotFound />} />
</Routes>
```

and nav links like this:
```
<NavLink
  href="/"
  end
>
  Home
</NavLink>
<NavLink
  href="/practice"
  end
>
  Practice
</NavLink>
```

In 0.2.1, if I routed to `http://localhost//` (notice the two slashes at the end) everything worked fine. It would route to `http://localhost//#/` and `http://localhost//#/practice` no problem.

In 0.3.1 this behavior is substantially different. `http://localhost//#/` works just fine, but when routing to `http://localhost//#/practice` it does not match the route I have set.

I can only assume that users are running into this problem because of a typo (typing too many slashes at the end of the address bar?) but it would be great to have this situation still work as expected since I feel like there is a bug in 0.3.1 somewhere that is causing it to act differently than 0.2.1.

If there is a better way to deal with weirdly formatted urls like this, I am all ears as well.

**Comment by rturnq** [comment]
Ok, I see the issue. The fix for #86 didn't correctly handle cases where there are path segments (including empty ones) preceding the hash. This should be an easy enough fix.

**Comment by rturnq** [comment]
This should be resolved in 0.3.2
